realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -602,6 +603,28 @@ public void onResult(int count) {
                         }
                     }
                 }
+                File managementFolder = new File(realmFolder, realmFileName + management);
+                if (managementFolder.exists()) {
+                    filesToDelete = Arrays.asList(
+                            new File(managementFolder, realmFileName + ".lock"),
+                            new File(managementFolder, realmFileName + ".log_a"),
+                            new File(managementFolder, realmFileName + ".log_b"),
+                            new File(managementFolder, realmFileName + ".log"));
+                    for (File fileToDelete : filesToDelete) {
```

Comment:
this logic is duplicated (with the legacy _Realm files_ above) can we have a 
```java
boolean deletes (File rootFolder, String realmFileName) {
// ...
}
```
then we can call it with 
```java
deletes(realmFolder, realmFileName);
```
and 
```java
deletes(managementFolder, realmFileName);
```